connection/connection.go

```diff
@@ -203,11 +210,30 @@ func (cmm ExtendedCSIMetricsManager) RecordMetricsClientInterceptor(
 	start := time.Now()
 	err := invoker(ctx, method, req, reply, cc, opts...)
 	duration := time.Since(start)
-	cmm.RecordMetrics(
-		method,   /* operationName */
-		err,      /* operationErr */
-		duration, /* operationDuration */
-	)
+
+	// record migration status
+	additionalInfo := ctx.Value(AdditionalInfoKey)
+	migrated := "false"
+	if additionalInfo != nil {
+		migrated = additionalInfo.(AdditionalInfo).Migrated
+	}
+	cmmv, metricsErr := cmm.WithLabelValues(map[string]string{metrics.LabelMigrated: migrated})
+	if metricsErr != nil {
+		klog.V(5).Infof("Failed to record migrated status, error: %v", metricsErr)
```

Comment:
IMHO this should cause the wrapper to fail. A log message with `V(5)` is likely to go unnoticed and this shouldn't happen in practice.

Reply:
Sorry I was adding comments to the wrong section. Now this should make sense. We have two cases here:
1. The metricsManager does not define labelMigration, we will fail the `WithLabelValues` with metricsErr and so we fall back to use the original metric. I dont know in this condition if we should log something... This can happen when a sidecar upgrade to the newest csi-lib-util version and did not change there MetricsManager initialization. What's your thought? 
2. The metricsManager has the labelMigration, the metricsErr will be nil, we use the wrapper to record the metric.

Reply:
I'm a little confused by this fallback behavior. Shouldn't we define "migrated" in the metric schema?

Reply:
> I'm a little confused by this fallback behavior. Shouldn't we define "migrated" in the metric schema?

That would also work... But that means we will have the migrated field in the MetricsManager by default and everything that is using it will have this new field, which I do not know if it is something we want... This current way of using `NewCSIMetricsManagerWithOptions(driverName, WithLabelNames(LabelMigrated))` is more like an opt-in way. So a sidecar/plugin can decide if they want this field or not...

Reply:
I change the HaveAdditionalLabel to public so hopefully the logic makes more sense here. 
So basically we will check if the metricsManager has the migrated label. If it does, it will record the migrated status, otherwise it will just use the old metrics.